### PR TITLE
Generate dedicated docker images for every service and finish setup of environment "testing"

### LINKED ISSUE #955: Create dedicated docker images for each service
Backend, frontend and offchain-resolver all use the generalized `backend`-image, which is huge and too complex. 
We already create a dedicated image for the delivery service. Apply the same approach to the three services mentioned above to improve composability and resource consumption.

### PR DESCRIPTION
**Pull request by malteish**
closes #955 

- creates several docker images on push
- fixes some stuff that wasn't set up correctly in the repo, like missing dependencies.

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -1,7 +1,70 @@
...
+RUN mkdir /library_archives
+RUN for lib in `ls -I billboard-api`; do cd $lib; yarn pack --filename /library_archives/$lib.tgz; cd ..;  done
+
+## prepare dependency operations for all packages
+WORKDIR '/build/packages'
+# several steps are performed here for each package individually (but not for the libraries):
+# 1. extract the information which libraries are needed for this package and store it in required_libraries
+# 2. remove all dependencied referencing "workspace" in the monorepo from package.json
+# 3. update the package.json with the resolutions from the docker/resolutions.json file
+RUN for package in `ls -I lib`;\
```
Comment (AlexNi245) [inline docker/Dockerfile:23]: Can you add a tiny comment above each bash line to explain what it does? That would increase the readability a lot

File: docker/Dockerfile [hunk 2]
```diff
@@ -1,7 +1,70 @@
-FROM --platform=linux/amd64 node:22-alpine
-WORKDIR /app
+FROM --platform=linux/amd64 node:22-alpine as build
```
Comment (AlexNi245) [inline docker/Dockerfile:1]: Isn't this file about to be removed?

File: packages/delivery-service/src/delivery.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import { Acknoledgment, getMessages, schema } from '@dm3-org/dm3-lib-delivery';
 import { auth } from '@dm3-org/dm3-lib-server-side';
 import { validateSchema } from '@dm3-org/dm3-lib-shared';
-import { getConversationId } from '@dm3-org/dm3-lib-storage';
+import { getConversationId } from '@dm3-org/dm3-lib-delivery';
```
Comment (AlexNi245) [inline packages/delivery-service/src/delivery.ts:4]: why has this moved to delivery?

**Review by AlexNi245 — CHANGES_REQUESTED** [review]

File: packages/delivery-service/src/delivery.ts (hunk 3 above)
Comment (malteish) [reply packages/delivery-service/src/delivery.ts:4]: It was actually already in dm3-lib-delivery, and also in lib-storage. As the delivery service needs the function, but doesn't use lib storage, I pointed it to lib-delivery. 
Then, afaik, getConversationId in lib-storage was not used anymore, so I removed it.

File: docker/Dockerfile (hunk 2 above)
Comment (malteish) [reply docker/Dockerfile]: true. removed it. 41838f3

File: packages/delivery-service/src/delivery.ts (hunk 3 above)
Comment (malteish) [reply packages/delivery-service/src/delivery.ts:4]: image
Also, it is only used in ds atm

File: docker/Dockerfile (hunk 1 above)
Comment (malteish) [reply docker/Dockerfile]: done 23e9b1a

**Pull request merged**